modules/gapi/samples/face_detection.cpp

```diff
@@ -0,0 +1,764 @@
+         const cv::GArrayDesc&,
+         const cv::GArrayDesc&,
+         double) {
+         return cv::empty_array_desc();
+    }
+};
+
+G_API_OP(SwapFaces,
+         <GFaces(GFaces)>,
+         "sample.custom.mtcnn.swap_faces") {
+          static cv::GArrayDesc outMeta(const cv::GArrayDesc&) {
+          return cv::empty_array_desc();
+    }
+};
+
+G_API_OP(Transpose,
+         <cv::GMat(cv::GMat)>,
+         "sample.custom.mtcnn.transpose") {
+          static cv::GMatDesc outMeta(const cv::GMatDesc in) {
+          const cv::GMatDesc out_desc = { in.depth, in.chan, cv::Size(in.size.height,
+                                                             in.size.width) };
+          return out_desc;
+    }
+};
+
+//Custom kernels implementation
+GAPI_OCV_KERNEL(OCVBuildFaces, BuildFaces) {
+    static void run(const cv::Mat & in_scores,
+        const cv::Mat & in_regresssions,
+        double scaleFactor,
+        double threshold,
+        std::vector<Face> &out_faces) {
+        out_faces = buildFaces(in_scores, in_regresssions, scaleFactor, threshold);
+    }
+};// GAPI_OCV_KERNEL(BuildFaces)
+
+GAPI_OCV_KERNEL(OCVRunNMS, RunNMS) {
+    static void run(const std::vector<Face> &in_faces,
+        double threshold,
+        bool useMin,
+        std::vector<Face> &out_faces) {
+        std::vector<Face> in_faces_copy = in_faces;
+        out_faces = Face::runNMS(in_faces_copy, threshold, useMin);
+    }
+};// GAPI_OCV_KERNEL(RunNMS)
+
+GAPI_OCV_KERNEL(OCVAccumulatePyramidOutputs, AccumulatePyramidOutputs) {
+    static void run(const std::vector<Face> &total_faces,
+        const std::vector<Face> &in_faces,
+        std::vector<Face> &out_faces) {
+        out_faces = total_faces;
+        out_faces.insert(out_faces.end(), in_faces.begin(), in_faces.end());
+    }
+};// GAPI_OCV_KERNEL(AccumulatePyramidOutputs)
+
+GAPI_OCV_KERNEL(OCVApplyRegression, ApplyRegression) {
+    static void run(const std::vector<Face> &in_faces,
+        bool addOne,
+        std::vector<Face> &out_faces) {
+        std::vector<Face> in_faces_copy = in_faces;
+        Face::applyRegression(in_faces_copy, addOne);
+        out_faces.clear();
+        out_faces.insert(out_faces.end(), in_faces_copy.begin(), in_faces_copy.end());
+    }
+};// GAPI_OCV_KERNEL(ApplyRegression)
+
+GAPI_OCV_KERNEL(OCVBBoxesToSquares, BBoxesToSquares) {
+    static void run(const std::vector<Face> &in_faces,
+        std::vector<Face> &out_faces) {
+        std::vector<Face> in_faces_copy = in_faces;
+        Face::bboxes2Squares(in_faces_copy);
+        out_faces.clear();
+        out_faces.insert(out_faces.end(), in_faces_copy.begin(), in_faces_copy.end());
+    }
+};// GAPI_OCV_KERNEL(BBoxesToSquares)
+
+GAPI_OCV_KERNEL(OCVR_O_NetPreProcGetROIs, R_O_NetPreProcGetROIs) {
+    static void run(const std::vector<Face> &in_faces,
+        const cv::Size & in_image_size,
+        std::vector<cv::Rect> &outs) {
+        outs.clear();
+        for (auto& f : in_faces) {
```

Comment:
Proposal: `for (const auto& face : in_faces)` instead current.

Reply:
Done